src/test/java/com/conveyal/gtfs/GTFSTest.java

```diff
@@ -605,6 +607,51 @@ private boolean runIntegrationTestOnFolder(
         );
     }
 
+    /**
+     * Load a feed and remove all data from the tables that match the mandatory file list. Confirm that the export
+     * contains the mandatory files which will be exported even though the matching table has no data.
+     */
+    @Test
+    void canExportEmptyMandatoryFiles() {
+        String testDBName = TestUtils.generateNewDB();
+        File tempFile = null;
+        try {
+            String zipFileName = TestUtils.zipFolderFiles("fake-agency", true);
+            String dbConnectionUrl = String.join("/", JDBC_URL, testDBName);
+            DataSource dataSource = TestUtils.createTestDataSource(dbConnectionUrl);
+            FeedLoadResult loadResult = GTFS.load(zipFileName, dataSource);
+            String namespace = loadResult.uniqueIdentifier;
+
+            // Remove data from tables that match the mandatory files.
+            for (String fileName : JdbcGtfsExporter.mandatoryFileList) {
+                try (Connection connection = dataSource.getConnection()) {
+                    String tableName = fileName.split("\\.")[0];
+                    String sql = String.format("delete from %s.%s", namespace, tableName);
+                    LOG.info(sql);
+                    connection.prepareStatement(sql).execute();
+                }
+            }
+
+            // Confirm that the mandatory files are present in the zip file.
+            tempFile = exportGtfs(namespace, dataSource, false);
+            ZipFile gtfsZipFile = new ZipFile(tempFile.getAbsolutePath());
+            for (String fileName : JdbcGtfsExporter.mandatoryFileList) {
+                Assert.assertNotNull(gtfsZipFile.getEntry(fileName));
+            }
+        } catch (IOException | SQLException e) {
+            tidyUpEmptyMandatoryFilesTest(testDBName, tempFile);
+        }
+        tidyUpEmptyMandatoryFilesTest(testDBName, tempFile);
```

Comment:
Combine these in the `finally` block. You may be able to inline the function as a result.

Reply:
@binh-dam-ibigroup thanks, have addressed.